--- rspamd-1.3.5/contrib/mumhash/mum.h.omv~	2016-09-17 14:41:27.204327763 +0200
+++ rspamd-1.3.5/contrib/mumhash/mum.h	2016-09-17 14:43:59.504276584 +0200
@@ -405,7 +405,9 @@ mum_hash (const void *key, size_t len, u
   if (avx2_support > 0)
     return _mum_hash_avx2 (key, len, seed);
   else if (! avx2_support) {
+#ifndef __clang__
     __builtin_cpu_init ();
+#endif
     avx2_support =  __builtin_cpu_supports ("avx2") ? 1 : -1;
     if (avx2_support > 0)
       return _mum_hash_avx2 (key, len, seed);
--- rspamd-1.3.5/contrib/mumhash/mum.h.omv~	2016-09-17 19:46:35.823702910 +0200
+++ rspamd-1.3.5/contrib/mumhash/mum.h	2016-09-17 19:46:38.767662622 +0200
@@ -114,7 +114,7 @@ _mum (uint64_t v, uint64_t p) {
      function doing 128x128->128 bit multiplication.  The function is
      very slow.  */
   lo = v * p, hi;
-  asm ("umulh %0, %1, %2" : "=r" (hi) : "r" (v), "r" (p));
+  __asm ("umulh %0, %1, %2" : "=r" (hi) : "r" (v), "r" (p));
 #else
   __uint128_t r = (__uint128_t) v * (__uint128_t) p;
   hi = (uint64_t) (r >> 64);
