diff -up rspamd-3.9.0/contrib/aho-corasick/acism.c.2~ rspamd-3.9.0/contrib/aho-corasick/acism.c
--- rspamd-3.9.0/contrib/aho-corasick/acism.c.2~	2024-07-12 20:58:29.101071644 +0200
+++ rspamd-3.9.0/contrib/aho-corasick/acism.c	2024-07-12 21:08:08.903715927 +0200
@@ -40,7 +40,7 @@
 
 #define BACK ((SYMBOL) 0)
 #define ROOT ((STATE) 0)
-extern const unsigned char lc_map[256];
+const unsigned char lc_map[256];
 
 int acism_lookup(ac_trie_t const *psp, const char *text, size_t len,
 				 ACISM_ACTION *cb, void *context, int *statep, bool caseless)
