diff -up rspamd-3.1/contrib/aho-corasick/acism.c.2~ rspamd-3.1/contrib/aho-corasick/acism.c
--- rspamd-3.1/contrib/aho-corasick/acism.c.2~	2021-11-01 15:33:30.000000000 +0100
+++ rspamd-3.1/contrib/aho-corasick/acism.c	2021-12-30 04:02:25.402543401 +0100
@@ -24,7 +24,7 @@
 
 #define BACK ((SYMBOL)0)
 #define ROOT ((STATE) 0)
-extern const guchar lc_map[256];
+const guchar lc_map[256];
 
 int
 acism_lookup(ac_trie_t const *psp, const char *text, size_t len,
