diff -up rspamd-3.7.4/src/libserver/rspamd_control.c.omv~ rspamd-3.7.4/src/libserver/rspamd_control.c
--- rspamd-3.7.4/src/libserver/rspamd_control.c.omv~	2023-11-14 17:32:46.550152895 +0100
+++ rspamd-3.7.4/src/libserver/rspamd_control.c	2023-11-14 17:33:40.105690354 +0100
@@ -921,6 +921,7 @@ rspamd_srv_handler(EV_P_ ev_io *w, int r
 					rdata->fd = cmd.cmd.spair.pair_num ? spair[1] : spair[0];
 				}
 				break;
+#ifdef WITH_HYPERSCAN
 			case RSPAMD_SRV_HYPERSCAN_LOADED:
 				/* Load RE cache to provide it for new forks */
 				if (rspamd_re_cache_is_hs_loaded(rspamd_main->cfg->re_cache) != RSPAMD_HYPERSCAN_LOADED_FULL ||
@@ -946,6 +947,7 @@ rspamd_srv_handler(EV_P_ ev_io *w, int r
 				rspamd_control_broadcast_cmd(rspamd_main, &wcmd, rfd,
 											 rspamd_control_ignore_io_handler, NULL, worker->pid);
 				break;
+#endif
 			case RSPAMD_SRV_MONITORED_CHANGE:
 				/* Broadcast command to all workers */
 				memset(&wcmd, 0, sizeof(wcmd));
