diff -up rspamd-3.0/contrib/doctest/doctest/doctest.h.omv~ rspamd-3.0/contrib/doctest/doctest/doctest.h
--- rspamd-3.0/contrib/doctest/doctest/doctest.h.omv~	2021-08-20 00:48:54.114889184 +0200
+++ rspamd-3.0/contrib/doctest/doctest/doctest.h	2021-08-20 00:53:26.576644476 +0200
@@ -4296,7 +4296,7 @@ namespace {
         static bool             isSet;
         static struct sigaction oldSigActions[DOCTEST_COUNTOF(signalDefs)];
         static stack_t          oldSigStack;
-        static char             altStackMem[4 * SIGSTKSZ];
+        static char*            altStackMem;
 
         static void handleSignal(int sig) {
             const char* name = "<unknown signal>";
@@ -4313,10 +4313,12 @@ namespace {
         }
 
         FatalConditionHandler() {
+            if(!altStackMem)
+                altStackMem = new char[4 * SIGSTKSZ];
             isSet = true;
             stack_t sigStack;
             sigStack.ss_sp    = altStackMem;
-            sigStack.ss_size  = sizeof(altStackMem);
+            sigStack.ss_size  = 4 * SIGSTKSZ;
             sigStack.ss_flags = 0;
             sigaltstack(&sigStack, &oldSigStack);
             struct sigaction sa = {};
@@ -4344,7 +4346,7 @@ namespace {
     bool             FatalConditionHandler::isSet                                      = false;
     struct sigaction FatalConditionHandler::oldSigActions[DOCTEST_COUNTOF(signalDefs)] = {};
     stack_t          FatalConditionHandler::oldSigStack                                = {};
-    char             FatalConditionHandler::altStackMem[]                              = {};
+    char*            FatalConditionHandler::altStackMem                                = nullptr;
 
 #endif // DOCTEST_PLATFORM_WINDOWS
 #endif // DOCTEST_CONFIG_POSIX_SIGNALS || DOCTEST_CONFIG_WINDOWS_SEH
